jina/serve/streamer.py

```diff
@@ -143,6 +155,52 @@ def stream(self, *args, **kwargs):
         """
         return self._streamer.stream(*args, **kwargs)
 
+    async def stream(
+        self,
+        docs: DocumentArray,
+        request_size: int = 100,
+        return_results: bool = False,
+        exec_endpoint: Optional[str] = None,
+        target_executor: Optional[str] = None,
+        parameters: Optional[Dict] = None,
+        results_in_order: bool = False,
+    ) -> AsyncIterator[Tuple[Union[DocumentArray, 'Request'], 'ExecutorError']]:
+        """
+        stream Documents and yield Documents or Responses and unpacked Executor error if any.
+
+        :param docs: The Documents to be sent to all the Executors
+        :param request_size: The amount of Documents to be put inside a single request.
+        :param return_results: If set to True, the generator will yield Responses and not `DocumentArrays`
+        :param exec_endpoint: The Executor endpoint to which to send the Documents
+        :param target_executor: A regex expression indicating the Executors that should receive the Request
+        :param parameters: Parameters to be attached to the Requests
+        :param results_in_order: return the results in the same order as the request_iterator
+        :yield: tuple of Documents or Responses and unpacked error from Executors if any
+        """
+        async for result in self.stream_docs(
+            docs=docs,
+            request_size=request_size,
+            return_results=True,  # force return Responses
+            exec_endpoint=exec_endpoint,
+            target_executor=target_executor,
+            parameters=parameters,
+            results_in_order=results_in_order,
+        ):
+            error = None
+            if jina_pb2.StatusProto.ERROR == result.status.code:
+                exception = result.status.exception
+                error = ExecutorError(
+                    name=exception.name,
+                    args=exception.args,
+                    stacks=exception.stacks,
+                    executor=exception.executor,
+                )
+            if return_results:
+                yield result, error
+            else:
+                result.document_array_cls = DocumentArray
```

Comment:
why is this needed?

Reply:
This block is the same as the block in in the async [post](https://github.com/jina-ai/jina/blob/master/jina/clients/mixin.py#L387) method which also returns an async generator.

Reply:
@samsja do you know why is this needed?

Reply:
> result.document_array_cls = DocumentArray
this is needed to for docarray v2 so that the DataRequest can be serialize to protobuf using the correct class (i,e, the return schema from the executor)

Reply:
okk

Reply:
but since it is a resulst here and not a request I am not sure it is needed